pyomo/core/base/reference.py

```diff
@@ -8,7 +8,14 @@
 #  This software is distributed under the 3-clause BSD License.
 #  ___________________________________________________________________________
 
-import collections
+try:
+    # python3
+    from collections.abc import MutableMapping as collections_MutableMapping
+    from collections.abc import Set as collections_Set
+except:                                           #pragma:nocover
```

Comment:
We shouldn't need this `#pragma:nocover`, as the 2.7 tests should pick up this branch.  More generally, why are we using a try-except instead of `if six.PY3`?

Reply:
Initially, I wasn't sure if it was a Python 3 change or a late Python 3 change. I suspect using the six check would be okay.

Reply:
I just checked, and `MutableMapping` is in `collections.abc` at least as far back as Python 3.4.